sycl/doc/ControllingSYCLLanguageFeatures.md

```diff
@@ -0,0 +1,196 @@
+# Controlling SYCL/DPC++ language features
+
+This documents aims to describe a mechanism which is used to control (i.e.
+enable or disable) different SYCL/DPC++ language and library features, like
+version of SYCL/DPC++ standard to use, set of extensions to use, etc.
+
+## Controlling SYCL/DPC++ standard version
+
+In order to control the language standard to compile for, the following options
+can be used:
+
+`-sycl-std=<value>`, `--sycl-std=<value>`, `--sycl-std <value>`
+
+SYCL language standard to compile for.
+
+Possible values:
+
+- `2017`, `121`, `1.2.1`, `sycl-1.2.1`: corresponds to SYCL 1.2.1
+  specification, see [SYCL Registry] for more details and the specification
+  text.
+
+- `dpcpp-0.8`: corresponds to DPC++ version documented in oneAPI 0.8
+
+  Basically, `-sycl-std=dpcpp-0.8` implies support for SYCL 1.2.1 specification
+  and bunch of extension enabled.
+
+  See [oneAPI Specification] for more details and the specification text.
+
+[SYCL Registry]: https://www.khronos.org/registry/SYCL/
+[oneAPI Specification]: https://spec.oneapi.com/
+
+Note: `-sycl-std` **doesn't** imply any C++ standard version to be set,
+which means that some default value will be selected. If by some reason, that
+default version is not a desired one,  it is possible to change C++ version
+independently of SYCL/DPC++ standard version via specifying additional option:
+`-sycl-std=1.2.1 -std=c++14`, for example.
+
+If SYCL/DPC++ standard version and C++ standard version (either default value
+for the compiler or one which was explicitly set via `-std`) are incompatible,
+then it is expected to see compilation errors. Incompatible means that C++
+standard version is less than minimum required by SYCL/DPC++ standard.
+
+`-std=<value>`, `--std=<value>`, `--std <value>`
+
+One more way to specify SYCL/DPC++ standard version is to use a general clang
+option, which allows to specify language standard to compile for.
+
+Supported values (besides listed in clang documentation/help):
```

Comment:
This is done at purpose, as I don't think that each value from `-sycl-std` fits `-std` option. I.e. with `-sycl-std` option it is clear that we are talking about SYCL (or about DPC++) standard, while for `-std` it might be as well CUDA, C++, HIP, OpenCL, etc. For example, `2017` might be confusing: is it C++17 or SYCL 1.2.1?

Reply:
Yes only options starting with `sycl-` or `dpcpp-` should work for with `-std`.